compiler-cli/src/fs.rs

```diff
@@ -199,17 +199,17 @@ pub fn write_outputs_under(outputs: &[OutputFile], base: &Path) -> Result<(), Er
 }
 
 pub fn write_output_under(file: &OutputFile, base: &Path) -> Result<(), Error> {
-    let OutputFile { path, text } = file;
-    write(&base.join(path), text)
+    let OutputFile { path, data } = file;
+    write(&base.join(path), data)
 }
 
 pub fn write_output(file: &OutputFile) -> Result<(), Error> {
-    let OutputFile { path, text } = file;
-    write(path, text)
+    let OutputFile { path, data } = file;
+    write(path, data)
 }
 
-pub fn write(path: &Path, text: &str) -> Result<(), Error> {
-    write_bytes(path, text.as_bytes())
+pub fn write(path: &Path, data: &OutputFileData) -> Result<(), Error> {
```

Comment:
Could you revert the change to this function please?

Reply:
@lpil please take another look if it is fine now (just pushed).